src/storage.rs

```diff
@@ -664,23 +664,58 @@ fn find_cargo_registry(cfg: &PartialConfig) -> Result<CargoRegistry, VetError> {
     }
 }
 
+/// tables with these names will be rendered by TomlFormatter as inline tables,
+/// rather than dotted tables.
+fn always_render_inline(key: &str) -> bool {
+    key == "dependency-criteria"
+}
+
+struct TomlFormatter;
+impl toml_edit::visit_mut::VisitMut for TomlFormatter {
+    fn visit_table_mut(&mut self, node: &mut toml_edit::Table) {
+        // Hide unnecessary implicit table headers for tables containing
+        // only other tables. We don't do this for empty tables as otherwise
+        // they could be hidden.
+        if !node.is_empty() {
+            node.set_implicit(true);
+        }
+        for (k, v) in node.iter_mut() {
+            if !always_render_inline(&k) {
+                // Try to convert the value into either a table or an array of
+                // tables if it is currently an inline table or inline array of
+                // tables.
+                *v = std::mem::take(v)
+                    .into_table()
+                    .map(toml_edit::Item::Table)
+                    .unwrap_or_else(|i| i)
+                    .into_array_of_tables()
+                    .map(toml_edit::Item::ArrayOfTables)
+                    .unwrap_or_else(|i| i);
+            }
+            self.visit_item_mut(v);
+        }
+    }
+}
+
```

Comment:
should this code be in the serialize module?

Reply:
Sure, I can move it there. Perhaps the `{load,store}_{toml,json}` methods should move there too? I mostly put it here because it feels like an implementation detail of `store_toml` which is in this file.

Reply:
yeah now that you've made them operate on writers I think it makes sense for it all to be considered "serialization" code and not "storage" code

Reply:
yeah now that you've made them operate on writers I think it makes sense for it all to be considered "serialization" code and not "storage" code

Reply:
I ended up just moving the formatting logic out, as the other parts feel like they're more related to the storage strategy.